File: Data Structures & Algorithms/permutations/submission-0.cpp
class Solution {
public:
    vector<vector<int>> permute(vector<int>& nums) {
        vector<vector<int>> res;
        permuteHelper(nums, 0, res);
        return res;
    }
private:
    void permuteHelper(vector<int>& nums, int i, vector<vector<int>>& res) {
        int n = nums.size();
        if(i == n) {
            res.push_back(nums);
            return;
        }
        
        for(int j = i; j < n; j++) {
            swap(nums[i], nums[j]);
            permuteHelper(nums, i + 1, res);
            swap(nums[i], nums[j]);
        }
    }
};

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        z = set()
        i = 0
        for j in range(k):
            if j >= len(nums):
                break
            if nums[j] not in z:
                z.add(nums[j])
            else:
                return True
        for j in range(k, len(nums)):
            if nums[j] not in z:
                z.add(nums[j])
            else:
                return True
            z.remove(nums[j - k])
        return False
        